File: S4-Systems-for-Design-and-Impl/Labs/Lab11/ProiectCSharp/TransportModel/transport.model/Seat.cs
using TransportModel.transport.model;

namespace Transport.Model.transport.model
{
    public class Seat : Entity<long>
    {
        public Reservation reservation { get; set; }
        public long seatNumber { get; set; }

        public Seat(Reservation reservation, long seatNumber)
        {
            this.reservation = reservation;
            this.seatNumber = seatNumber;
        }

        public override string ToString()
        {
            return $"Seat{{reservation={reservation}, seatNumber={seatNumber}}}";
        }

        public override bool Equals(object obj)
        {
            if (this == obj) return true;
            if (!(obj is Seat)) return false;

            Seat seat = (Seat)obj;

            if (!reservation.Equals(seat.reservation)) return false;
            return seatNumber.Equals(seat.seatNumber);
        }

        public override int GetHashCode()
        {
            int result = reservation.GetHashCode();
            result = 31 * result + seatNumber.GetHashCode();
            return result;
        }
    }
}

File: S4-Systems-for-Design-and-Impl/Labs/Lab5/ProiectJavaNew/src/main/java/domain/Seat.java
package domain;

public class Seat extends Entity<Long> {

    private Reservation reservation;
    private Long seatNumber;

    public Seat(Reservation reservation, Long seatNumber) {
        this.reservation = reservation;
        this.seatNumber = seatNumber;
    }

    public Reservation getReservation() {
        return reservation;
    }

    public void setReservation(Reservation reservation) {
        this.reservation = reservation;
    }

    public Long getSeatNumber() {
        return seatNumber;
    }

    public void setSeatNumber(Long seatNumber) {
        this.seatNumber = seatNumber;
    }

    @Override
    public String toString() {
        return "Seat{" +
                "reservation=" + reservation +
                ", seatNumber=" + seatNumber +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Seat)) return false;

        Seat seat = (Seat) o;

        if (!reservation.equals(seat.reservation)) return false;
        return seatNumber.equals(seat.seatNumber);
    }

    @Override
    public int hashCode() {
        int result = reservation.hashCode();
        result = 31 * result + seatNumber.hashCode();
        return result;
    }
}
